internal/gcs/guestconnection.go

```diff
@@ -105,7 +105,9 @@ func (gc *GuestConnection) Protocol() uint32 {
 }
 
 // connect establishes a GCS connection. It must not be called more than once.
-func (gc *GuestConnection) connect(ctx context.Context) (err error) {
+// sendHostCreate should be set to `true` when connecting to the uvm that is created from
+// scratch. It should be `false` when connecting to the cloned uvm.
+func (gc *GuestConnection) connect(ctx context.Context, sendHostCreate bool) (err error) {
```

Comment:
I think it would be more meaningful to name this something like `isColdStart` rather than `sendHostCreate`. `sendHostCreate` describes the action we take because of the value, but it's more readable to use a name that describes the intent behind the value.